extensions/git/src/commands.ts

```diff
@@ -60,6 +60,26 @@ class CheckoutRemoteHeadItem extends CheckoutItem {
 	}
 }
 
+class BranchDeleteItem implements QuickPickItem {
+
+	protected get shortCommit(): string { return (this.ref.commit || '').substr(0, 8); }
+	protected get treeish(): string | undefined { return this.ref.name; }
+	get label(): string { return this.ref.name || this.shortCommit; }
+	get description(): string { return this.shortCommit; }
+
+	constructor(protected ref: Ref) { }
+
+	async run(model: Model): Promise<void> {
+		const ref = this.treeish;
+
+		if (!ref) {
+			return;
+		}
+
+		await model.deleteBranch(ref);
```

Comment:
A common error from this operation happens when the branch isn't yet merged:

![image](https://cloud.githubusercontent.com/assets/22350/25742311/24ce0144-318f-11e7-96ea-81dfbf6f3d5d.png)

It would be great if the underlying git library would catch this situation, wrap it neatly in a `GitError` with a new `GitErrorCode` and we `try catch` it here, prompting the user `Hey, the branch isn't yet merge. Do you still want to delete it?`.